mlflow/pyfunc/model.py

```diff
@@ -237,7 +238,7 @@ def _save_model_with_class_artifacts_params(
                 tmp_artifacts_config[artifact_name] = tmp_artifact_path
                 saved_artifact_subpath = posixpath.join(
                     saved_artifacts_dir_subpath,
-                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()),
+                    pathlib.Path(os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path())).as_posix(),
```

Comment:
If we're going with pathlib.Path() (which I think is a very good call here), should we use the native pathlib relative path syntax?

Something along the lines of:

```python
if artifacts:
    saved_artifacts_config = {}
    with TempDir() as tmp_artifacts_dir:
        tmp_artifacts_config = {}
        saved_artifacts_dir_subpath = "artifacts"
        for artifact_name, artifact_uri in artifacts.items():
            tmp_artifact_path = _download_artifact_from_uri(
                artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path()
            )
            tmp_artifacts_config[artifact_name] = tmp_artifact_path
            
            tmp_artifact_path_obj = Path(tmp_artifact_path)
            tmp_artifacts_dir_path_obj = Path(tmp_artifacts_dir.path())
            
            # use native pathlib operator
            relative_path = tmp_artifact_path_obj.relative_to(tmp_artifacts_dir_path_obj).as_posix()
            
            # Join paths using posixpath.join
            saved_artifact_subpath = posixpath.join(saved_artifacts_dir_subpath, relative_path)
            
            saved_artifacts_config[artifact_name] = {
                CONFIG_KEY_ARTIFACT_RELATIVE_PATH: saved_artifact_subpath,
                CONFIG_KEY_ARTIFACT_URI: artifact_uri,
            }
            
        shutil.move(tmp_artifacts_dir.path(), os.path.join(path, saved_artifacts_dir_subpath))
    custom_model_config_kwargs[CONFIG_KEY_ARTIFACTS] = saved_artifacts_config

```
It might make the code a bit easier to reason about as we'd no longer be mixing posixpath, pathlib, and os.path all in one path resolution logical block.